will-moore/omero-figure
# OME-Zarr support

OMERO.figure without OMERO!

Uses zarrita.js to load OME-Zarr data and manually renders to image src.

Work in progress...

To test:
 - Go to https://deploy-preview-5--omero-figure.netlify.app/
 - Click on `New File` and enter Zarr Url into the `Add Images` dialog. E.g. https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.4/idr0062A/6001240.zarr
 
![Screenshot 2024-12-09 at 10 45 07](https://github.com/user-attachments/assets/507504fb-0530-46ce-91c4-748ce1015a9d)



TODOs to support zarr images:

- [ ] Handle missing `omero.channels` when loading images - generate on the fly
- [ ] Handle `async get_img_src()`  everywhere
- [x] Add caching to avoid re-loading zarr data if possible
- [ ] Only render a region of images (for big images zoomed in)
- [ ] LUTs!

Other TODOs:
 - [ ] File > Open should ask for a file from URL or local file (when static deployed)
 - [ ] Support `figure/?file=url-to-figure.json`
 - [ ] File > Download... option (useful even without static deployment). With static deploy, File > Save should do this.
